src/node.rs

```diff
@@ -633,6 +641,27 @@ impl<S: std::fmt::Debug + ForkSource> InMemoryNodeInner<S> {
     pub fn stop_impersonating_account(&mut self, address: Address) -> bool {
         self.impersonated_accounts.remove(&address)
     }
+
+    /// Archives the current state for later queries.
+    pub fn archive_state(&mut self) -> Result<(), String> {
+        if self.previous_states.len() == MAX_PREVIOUS_STATES as usize {
+            if let Some(entry) = self.previous_states.shift_remove_index(0) {
+                log::info!("removing archived state for previous block {:#x}", entry.0);
```

Comment:
consider switching to debug (as this will get spammy - will print after every block after 128 blocks)